### PR TITLE
:books: Lobby Page

### PR DESCRIPTION
**Pull request by AlexandreMichelix**
## Description

- Implementing Game configuration Page
- Adding ShareLink widget
- Fix somes changes to correspond to figma model page

## Checklist before requesting a review

-   [x] I have performed a self-review of my code
-   [x] My pull request title respect the naming convention: [**_branch-name_**] **_summary_**
-   [x] I've added the corresponding label
-   [x] If my pull request implies some visual changes, I've linked screenshots in the description to show them
-   [x] If my pull request is linked to a task or a ticket, I've put the link in the description

---> [ClickUp](https://app.clickup.com/t/860rcrc8h)

image
image

### CODE REVIEW

File: lib/shared/ShareLink/share_link.dart [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  const ShareLinkWidget({super.key});
+
+  @override
+  State<ShareLinkWidget> createState() => _ShareLinkWidget();
+}
+
+class _ShareLinkWidget extends State<ShareLinkWidget> {
+  final TextEditingController textField = TextEditingController(
+      text:
+          'https://www.battleship.fr/game?id=8b1a292b05c946bc945b8d14a105dd5d');
```
Comment (DrakenLine) [inline lib/shared/ShareLink/share_link.dart:22]: why did you initialize the text value ? Is it not better to use `hintText` property ?

File: lib/shared/ShareLink/share_link.dart [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                icon: IconWidget(
+                    name: IconName.copy,
+                    size: IconSize.small,
+                    fillColor: BSHexaColors().white,
+                    borderColor: BSHexaColors().darkPurple2),
+                buttonType: ButtonType.secondary,
+                contentType: ContentType.icon,
+                function: () {
+                  if (textField.text.isNotEmpty) {
+                    Clipboard.setData(ClipboardData(text: textField.text));
```
Comment (DrakenLine) [inline lib/shared/ShareLink/share_link.dart:61]: why did you use `Clipboard` ?

**Review by DrakenLine — CHANGES_REQUESTED** [review]

File: lib/shared/ShareLink/share_link.dart (hunk 2 above)
Comment (AlexandreMichelix) [reply lib/shared/ShareLink/share_link.dart:61]: to copy pasta 🍝

File: android/app/build.gradle [hunk 3]
```diff
@@ -47,7 +47,7 @@ android {
         applicationId "com.example.battleship"
         // You can update the following values to match your application needs.
         // For more information, see: https://docs.flutter.dev/deployment/android#reviewing-the-build-configuration.
-        minSdkVersion flutter.minSdkVersion
+        minSdkVersion 19
```
Comment (FlavianEng) [inline android/app/build.gradle:50]: Why this change ?

File: lib/pages/authentication/authentication_form.page.dart [hunk 4]
```diff
@@ -210,6 +210,7 @@ class _AuthenticationFormPageState extends State<AuthenticationFormPage> {
                         key: emailForm,
                         child: InputWidget(
                           controller: emailField,
+                          readOnly: false,
```
Comment (FlavianEng) [inline lib/pages/authentication/authentication_form.page.dart:213]: Isn't it the default value ?

File: lib/pages/game_configuration/game_configuration.page.dart [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+}
+
+class _GameConfig extends State<GameConfigurationPage> {
+  final TextEditingController textField = TextEditingController(
+      text:
+          'https://www.battleship.fr/game?id=8b1a292b05c946bc945b8d14a105dd5d');
+
+  void update() {
+    setState(() {});
+  }
```
Comment (FlavianEng) [inline lib/pages/game_configuration/game_configuration.page.dart:30]: Not sure it's necessary to create this function

File: lib/pages/game_configuration/game_configuration.page.dart [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+                          return Dialog(
+                            backgroundColor: BSColors().darkPurple,
+                            shape: RoundedRectangleBorder(
+                              borderRadius: BorderRadius.circular(8.0),
+                            ),
+                            child: SizedBox(
+                              width: MediaQuery.of(context).size.width / 1,
+                              height: MediaQuery.of(context).size.width / 3.5,
+                              child:
+                                  const ShareLinkWidget(), // Insert your widget here
```
Comment (FlavianEng) [inline lib/pages/game_configuration/game_configuration.page.dart:122]: Remove comment

File: lib/shared/Container/container_widget.dart [hunk 7]
```diff
@@ -38,7 +29,10 @@ class _ContainerWidgetState extends State<ContainerWidget> {
                 )
               : const Border(),
         ),
-        child: widget.child,
+        child: SizedBox(
```
Comment (FlavianEng) [inline lib/shared/Container/container_widget.dart:32]: Shouldn't you use a [FractionallySizedBox](https://api.flutter.dev/flutter/widgets/FractionallySizedBox-class.html) widget ?

File: lib/shared/Header/header_widget.dart [hunk 8]
```diff
@@ -48,9 +50,28 @@ PreferredSize generateAppBar(BuildContext context,
                           child:
                               TitleWidget(title: title, type: TitleType.h1))))
               : Container(),
-          content != null ? Center(child: content) : Container()
+          content != null ? Center(child: content) : Container(),
```
Comment (FlavianEng) [inline lib/shared/Header/header_widget.dart:53]: Reverse condition, please. It's easier to your brain to think positively

File: lib/shared/Input/input_widget.dart [hunk 9]
```diff
@@ -37,16 +43,43 @@ class _InputWidgetState extends State<InputWidget> {
...
-          padding: const EdgeInsets.all(8.0),
-          child: Text(
-            widget.isRequired != true ? widget.title : '${widget.title} *',
-            style: BattleShipFont.policy.titleMedium!
-                .apply(color: BSColors().white),
-          ),
+          padding: const EdgeInsets.only(bottom: 10.0),
+          child: (() {
+            if (widget.title != null) {
+              if (widget.icon != null) {
```
Comment (FlavianEng) [inline lib/shared/Input/input_widget.dart:49]: Reverse condition, please. It's easier to your brain to think positively

File: lib/shared/Input/input_widget.dart [hunk 10]
```diff
@@ -104,7 +137,7 @@ class _InputWidgetState extends State<InputWidget> {
                 counterText: '',
                 filled: true,
                 contentPadding:
-                    const EdgeInsets.symmetric(vertical: 20, horizontal: 20),
+                    const EdgeInsets.symmetric(vertical: 10, horizontal: 10),
```
Comment (FlavianEng) [inline lib/shared/Input/input_widget.dart:140]:
```suggestion
                    const EdgeInsets.all(10),
```

File: lib/shared/PlayerCard/player_card.dart [hunk 11]
```diff
@@ -26,21 +26,31 @@ class _PlayerCardState extends State<PlayerCard> {
       mainAxisAlignment: MainAxisAlignment.center,
       children: <Widget>[
         SizedBox(
```
Comment (FlavianEng) [inline lib/shared/PlayerCard/player_card.dart:28]: Shouldn't you use [FractionallySizedBox widget](https://api.flutter.dev/flutter/widgets/FractionallySizedBox-class.html) ?

File: lib/shared/ShareLink/share_link.dart [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: [
+        Row(
+          mainAxisAlignment: MainAxisAlignment.center,
+          crossAxisAlignment: CrossAxisAlignment.end,
+          children: [
+            SizedBox(
```
Comment (FlavianEng) [inline lib/shared/ShareLink/share_link.dart:33]: Shouldn't you use [FractionallySizedBox widget](https://api.flutter.dev/flutter/widgets/FractionallySizedBox-class.html) ?

File: lib/shared/Versus/versus_widget.dart [hunk 13]
```diff
@@ -15,35 +16,34 @@ class _VersusWidgetState extends State<VersusWidget> {
   @override
   Widget build(BuildContext context) {
     return SizedBox(
```
Comment (FlavianEng) [inline lib/shared/Versus/versus_widget.dart:18]: Shouldn't you use [FractionallySizedBox widget](https://api.flutter.dev/flutter/widgets/FractionallySizedBox-class.html) ?

File: android/app/build.gradle (hunk 3 above)
Comment (AlexandreMichelix) [reply android/app/build.gradle:50]: image

File: lib/shared/Button/button_widget.dart [hunk 14]
```diff
@@ -218,12 +208,13 @@ class _ButtonWidget extends State<ButtonWidget> {
...
-                  height: 8,
-                  width: 8,
+                  height: 6,
+                  width: 6,
                 ),
               ),
               Padding(
-                  padding: getPaddingSize(),
+                  padding:
+                      EdgeInsets.symmetric(horizontal: screenWidth * 0.018),
```
[the remark this answers is not in the document]
Comment (AlexandreMichelix) [reply lib/shared/Button/button_widget.dart:217]: the perfect size of width button widget depending on every screen

File: lib/pages/game_configuration/game_configuration.page.dart [hunk 15]
```diff
@@ -0,0 +1,137 @@
...
+        children: [
+          Row(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: [
+              Column(
+                children: [
+                  Container(
+                    padding: const EdgeInsets.all(10),
+                    child: const ContainerWidget(
+                      padding: 20,
```
[the remark this answers is not in the document]
Comment (AlexandreMichelix) [reply lib/pages/game_configuration/game_configuration.page.dart:60]: image
ContainerWidget is customized and the params is already an edge inset

File: lib/shared/ShareLink/share_link.dart (hunk 12 above)
Comment (AlexandreMichelix) [reply lib/shared/ShareLink/share_link.dart:33]: In this widget no

File: lib/pages/authentication/authentication_form.page.dart (hunk 4 above)
Comment (MaximeWrlz) [reply lib/pages/authentication/authentication_form.page.dart:213]: So what was the solution ?

File: lib/shared/ShareLink/share_link.dart (hunk 1 above)
Comment (MaximeWrlz) [reply lib/shared/ShareLink/share_link.dart:22]: I don't understand... if your branch is based on mine, you should have the rights default text value, which is "Creating game..."

File: lib/shared/PlayerCard/player_card.dart (hunk 11 above)
Comment (AlexandreMichelix) [reply lib/shared/PlayerCard/player_card.dart:28]: heightfactor and widthfactor didn't work

File: lib/shared/Versus/versus_widget.dart (hunk 13 above)
Comment (AlexandreMichelix) [reply lib/shared/Versus/versus_widget.dart:18]: same as before

File: macos/Flutter/ephemeral/flutter_export_environment.sh [hunk 16]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+# This is a generated file; do not edit or check into version control.
+export "FLUTTER_ROOT=/home/alexandre/snap/flutter/common/flutter"
+export "FLUTTER_APPLICATION_PATH=/home/alexandre/Documents/Ynov/projet_fil_rouge/battleship_mobile"
```
Comment (MaximeWrlz) [inline macos/Flutter/ephemeral/flutter_export_environment.sh:4]: The fact that the relative path is connected to your pc will not be a problem?

File: macos/Flutter/ephemeral/Flutter-Generated.xcconfig [hunk 17]
```diff
@@ -0,0 +1,11 @@
+// This is a generated file; do not edit or check into version control.
+FLUTTER_ROOT=/home/alexandre/snap/flutter/common/flutter
+FLUTTER_APPLICATION_PATH=/home/alexandre/Documents/Ynov/projet_fil_rouge/battleship_mobile
```
Comment (MaximeWrlz) [inline macos/Flutter/ephemeral/Flutter-Generated.xcconfig:3]: The fact that the relative path is connected to your pc will not be a problem?

File: lib/shared/ShareLink/share_link.dart (hunk 12 above)
Comment (MaximeWrlz) [reply lib/shared/ShareLink/share_link.dart:33]: Maybe it could improve responsive on specific mobile screens

**Review by MaximeWrlz — CHANGES_REQUESTED** [review]

File: lib/shared/Input/input_widget.dart [hunk 18]
```diff
@@ -45,8 +45,18 @@ class _InputWidgetState extends State<InputWidget> {
         Padding(
           padding: const EdgeInsets.only(bottom: 10.0),
           child: (() {
-            if (widget.title != null) {
-              if (widget.icon != null) {
+            if (widget.title == null) {
```
Comment (MaximeWrlz) [inline lib/shared/Input/input_widget.dart:48]: You can't do domething like this: `if (widget.title) {...}` ?

**Review by MaximeWrlz — APPROVED** [review]

File: lib/shared/Input/input_widget.dart (hunk 18 above)
Comment (AlexandreMichelix) [reply lib/shared/Input/input_widget.dart:48]: Conditions must have a static type of 'bool'

File: android/app/build.gradle (hunk 3 above)
Comment (FlavianEng) [reply android/app/build.gradle:50]: Change flutter minSdkVersion instead -> https://www.flutterbeads.com/change-android-minsdkversion-in-flutter/?expand_article=1

File: lib/shared/Button/button_widget.dart (hunk 14 above)
[the remark this answers is not in the document]
Comment (FlavianEng) [reply lib/shared/Button/button_widget.dart:217]: Based on what ?

File: android/app/build.gradle [hunk 19]
```diff
@@ -47,7 +47,7 @@ android {
         applicationId "com.example.battleship"
         // You can update the following values to match your application needs.
         // For more information, see: https://docs.flutter.dev/deployment/android#reviewing-the-build-configuration.
-        minSdkVersion 19
+        minSdkVersion localProperties.getProperty('flutter.minSdkVersion').toInteger()
```
Comment (FlavianEng) [inline android/app/build.gradle:50]: wtf why not this :
```suggestion
        minSdkVersion flutter.minSdkVersion
```

File: .gitignore [hunk 20]
```diff
@@ -44,5 +44,8 @@ app.*.map.json
 /android/app/profile
 /android/app/release
 
+# Generated files for Device mobile app
+**/macos/Flutter
+
```
Comment (FlavianEng) [inline .gitignore:49]: Check with @DrakenLine

**Review by DrakenLine — APPROVED** [review]

**Pull request merged**